packages/fxa-auth-server/lib/oauth/jwt_id_token.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AppError = require('./error');
+const jwt = require('./jwt');
+
+const HEADER_TYP = 'at+JWT';
+
+/**
+ * Verify an OIDC ID Token, following the flow in 3.1.3.7 of the spec:
+ * https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation
+ *
+ * @param {String} idToken
+ * @param {String} clientId
+ * @throws `invalidToken` error if ID Token is invalid.
+ * @returns {Promise<Object>} resolves with claims in Token
+ */
+exports.verify = async function verify(idToken, clientId) {
+  // Steps 1 and 2 of the verification flow are already handled by jwt.js:
+  //
+  // 1. if the ID Token is encrypted, decrypt it using the keys and algorithms
+  //    negotiated at Registration time between OpenID Provider and Client.
+  //    If encryption was negotiated but the token is not encrypted, RP SHOULD
+  //    reject it.
```

Comment:
We don't support encrypted JWTs, it may be worth adding a note to that effect in here so that future readers don't go looking for that code.